internal/experiment/openvpn/endpoint.go

```diff
@@ -239,42 +217,17 @@ func getOpenVPNConfig(
 				Port:   endpoint.Port,
 				Proto:  vpnconfig.Proto(endpoint.Transport),
 
-				// options coming from the default known values.
-				Cipher: baseOptions.Cipher,
-				Auth:   baseOptions.Auth,
-
-				// auth coming from passed credentials.
-				CA:   creds.CA,
-				Cert: creds.Cert,
-				Key:  creds.Key,
+				// options and credentials come from the experiment
+				// richer input targets.
+				Cipher: config.Cipher,
+				Auth:   config.Auth,
+				CA:     []byte(config.SafeCA),
+				Cert:   []byte(config.SafeCert),
+				Key:    []byte(config.SafeKey),
 			},
 		),
 		vpnconfig.WithHandshakeTracer(tracer),
 	)
 
 	return cfg, nil
 }
-
-// maybeExtractBase64Blob is used to pass credentials as command-line options.
-func maybeExtractBase64Blob(val string) (string, error) {
-	s := strings.TrimPrefix(val, "base64:")
-	if len(s) == len(val) {
-		// no prefix, so we'll treat this as a pem-encoded credential.
-		return s, nil
-	}
-	dec, err := base64.URLEncoding.DecodeString(strings.TrimSpace(s))
-	if err != nil {
-		return "", fmt.Errorf("%w: %s", ErrBadBase64Blob, err)
-	}
-	return string(dec), nil
-}
```

Comment:
>  I assume your original intent for having a base64-encoded blob was that this would have made it easier for people to specify the option from the command line, right? 

it is easier for me to copy&paste and print, and I think it gave a more compact blob. API is returning the raw PEM though, and I preferred to remove all the boilerplate.

from cli it should work with encoding the newlines etc. I think it's true that at least *some* options would better come from an oonirun json.

fwiw, I found useful to develop to just pass a cli flag to a non-json specific config file (i.e., an OpenVPN or wireguard config file). I've removed those for style, but some times there's extra work to format oonirun when perhaps other format is more suitable to experiment. OTOH, for production I'd rather avoid free inputs...

Reply:
> fwiw, I found useful to develop to just pass a cli flag to a non-json specific config file (i.e., an OpenVPN or wireguard config file). I've removed those for style, but some times there's extra work to format oonirun when perhaps other format is more suitable to experiment. OTOH, for production I'd rather avoid free inputs...

We can imagine an even further future in which the experiment parses its own command line options and decides what to do with them. This would be the end of the radical refactoring where we're basically moving as much as possible to experiments and dependent packages to empty to the core libraries. In such a case, I think it would definitely make lots of sense to have a way to provide an OpenVPN config or equivalent to an experiment.

Reply:
I am going to close the conversation. Please, reopen if you want to continue discussing here.